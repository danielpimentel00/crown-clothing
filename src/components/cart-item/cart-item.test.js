import { shallow } from "enzyme";
import CartItem from "./cart-item.component";

describe("Cart Item Component", () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      name: "tShirt",
      imageUrl: "https://www.sss.com",
      quantity: 0,
      price: 50,
    };
    wrapper = shallow(<CartItem cartItem={mockProps} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("check if price is right", () => {
    const quantity = 5;
    const price = 100;
    mockProps.quantity = quantity;
    mockProps.price = price;
    const wrapper2 = shallow(<CartItem cartItem={mockProps} />);

    expect(wrapper2.find(".price").getElement().props.children).toEqual([
      quantity,
      " x $",
      price,
    ]);
  });
});
