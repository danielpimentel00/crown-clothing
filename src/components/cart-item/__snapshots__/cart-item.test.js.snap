// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Item Component should match snapshot 1`] = `ShallowWrapper {}`;

exports[`cart-item component should match snapshot 1`] = `ShallowWrapper {}`;
